pkg/proxy/iptables/proxier.go

```diff
@@ -134,6 +134,43 @@ func (lkct LinuxKernelCompatTester) IsCompatible() error {
 const sysctlRouteLocalnet = "net/ipv4/conf/all/route_localnet"
 const sysctlBridgeCallIPTables = "net/bridge/bridge-nf-call-iptables"
 
+// Since converting probabilities (floats) to strings is expensive
+// and we are using only probabilities in the format of 1/n, we are
+// precomputing some number of those and cache for future reuse.
+var precomputedProbabilities []string
+
+func init() {
+	const initiallyPrecomputed = 1000
+	precomputeProbabilities(initiallyPrecomputed)
+}
+
+func precomputeProbabilities(numberOfPrecomputed int) {
+	result := make([]string, numberOfPrecomputed+1)
+	copied := copy(result, precomputedProbabilities)
+	result[0] = "<bad value>"
+	for i := copied; i <= numberOfPrecomputed; i++ {
+		result[i] = computeProbability(i)
+	}
+	precomputedProbabilities = result
+}
+
+func computeProbability(n int) string {
+	return fmt.Sprintf("%0.5f", 1.0/float64(n))
+}
+
+func probability(n int) string {
+	if n >= len(precomputedProbabilities) {
+		// We ensure that we grow the slice of precomputed probabilties
+		// at least by 2x, to have amortized cost of copying values.
+		newNumberOfPrecomputed := n
+		if newNumberOfPrecomputed < 2*len(precomputedProbabilities) {
```

Comment:
Done (it wasn't only about allocating here, but also precomputing, but I agree it doesn't make that much sense).

Should be fixed now.